pkg/rke2/rke2_linux.go

```diff
@@ -192,6 +202,188 @@ func initExecutor(clx *cli.Context, cfg Config, isServer bool) (*podexecutor.Sta
+				PeriodSeconds:       &controlPlaneProbes.Etcd.Liveness.PeriodSeconds,
+			},
+			Readiness: {
+				InitialDelaySeconds: &controlPlaneProbes.Etcd.Readiness.InitialDelaySeconds,
+				TimeoutSeconds:      &controlPlaneProbes.Etcd.Readiness.TimeoutSeconds,
+				FailureThreshold:    &controlPlaneProbes.Etcd.Readiness.FailureThreshold,
+				PeriodSeconds:       &controlPlaneProbes.Etcd.Readiness.PeriodSeconds,
+			},
+			Startup: {
+				InitialDelaySeconds: &controlPlaneProbes.Etcd.Startup.InitialDelaySeconds,
+				TimeoutSeconds:      &controlPlaneProbes.Etcd.Startup.TimeoutSeconds,
+				FailureThreshold:    &controlPlaneProbes.Etcd.Startup.FailureThreshold,
+				PeriodSeconds:       &controlPlaneProbes.Etcd.Startup.PeriodSeconds,
+			},
+		},
+		CloudControllerManager: {
+			Liveness: {
+				InitialDelaySeconds: &controlPlaneProbes.CloudControllerManager.Liveness.InitialDelaySeconds,
+				TimeoutSeconds:      &controlPlaneProbes.CloudControllerManager.Liveness.TimeoutSeconds,
+				FailureThreshold:    &controlPlaneProbes.CloudControllerManager.Liveness.FailureThreshold,
+				PeriodSeconds:       &controlPlaneProbes.CloudControllerManager.Liveness.PeriodSeconds,
+			},
+			Readiness: {
+				InitialDelaySeconds: &controlPlaneProbes.CloudControllerManager.Readiness.InitialDelaySeconds,
+				TimeoutSeconds:      &controlPlaneProbes.CloudControllerManager.Readiness.TimeoutSeconds,
+				FailureThreshold:    &controlPlaneProbes.CloudControllerManager.Readiness.FailureThreshold,
+				PeriodSeconds:       &controlPlaneProbes.CloudControllerManager.Readiness.PeriodSeconds,
+			},
+			Startup: {
+				InitialDelaySeconds: &controlPlaneProbes.CloudControllerManager.Startup.InitialDelaySeconds,
+				TimeoutSeconds:      &controlPlaneProbes.CloudControllerManager.Startup.TimeoutSeconds,
+				FailureThreshold:    &controlPlaneProbes.CloudControllerManager.Startup.FailureThreshold,
+				PeriodSeconds:       &controlPlaneProbes.CloudControllerManager.Startup.PeriodSeconds,
+			},
+		},
+	}
+
+	// defaultProbeConf contains a map of default probe settings for each type, used if not explicitly configured.
+	var defaultProbeConf = map[string]map[string]int32{
+		// https://github.com/kubernetes/kubernetes/blob/v1.24.0/cmd/kubeadm/app/util/staticpod/utils.go#L246
+		Liveness: {
+			InitialDelaySeconds: 10,
+			TimeoutSeconds:      15,
+			FailureThreshold:    8,
+			PeriodSeconds:       10,
+		},
+		// https://github.com/kubernetes/kubernetes/blob/v1.24.0/cmd/kubeadm/app/util/staticpod/utils.go#L252
+		Readiness: {
+			InitialDelaySeconds: 0,
+			TimeoutSeconds:      15,
+			FailureThreshold:    3,
+			PeriodSeconds:       1,
+		},
+		// https://github.com/kubernetes/kubernetes/blob/v1.24.0/cmd/kubeadm/app/util/staticpod/utils.go#L259
+		Startup: {
+			InitialDelaySeconds: 10,
+			TimeoutSeconds:      5,
+			FailureThreshold:    24,
+			PeriodSeconds:       10,
+		},
+	}
+
+	var parsedProbeConf = make(map[string]int32)
+
+	if cfg.ControlPlaneProbeConf != "" {
```

Comment:
Could you convert 358-380 into a separate function? and then write a unit test for that function?

Reply:
Honestly this whole function has sort of grown out of control. The bits for resources, environment variables, mounts, and probe configs probably all merit going into their own functions.